src/assets/scripts/client/aircraft/AircraftInstanceModel.js

```diff
@@ -2272,49 +1259,146 @@ export default class Aircraft {
             return;
         }
 
+        this.updateAircraftTurnPhysics();
+
+        this.updateAltitudePhysics();
+
+        if (this.isOnGround()) {
+            this.trend = 0;
+        }
+
+        // SPEED
+        this.updateSpeedPhysics();
+
+        if (!this.position) {
+            return;
+        }
+
+        // Trailling
+        if (this.position_history.length === 0) {
+            this.position_history.push([
+                this.position[0],
+                this.position[1],
+                window.gameController.game_time() / window.gameController.game_speedup()
+            ]);
+            // TODO: this can be abstracted
+        } else if (abs((window.gameController.game_time() / window.gameController.game_speedup()) - this.position_history[this.position_history.length - 1][2]) > 4 / window.gameController.game_speedup()) {
+            this.position_history.push([this.position[0], this.position[1], window.gameController.game_time() / window.gameController.game_speedup()]);
+        }
+
+        // FIXME: is this ratio correct? is it 0.000514444 or 0.514444?
+        const scaleSpeed = this.speed * 0.000514444 * window.gameController.game_delta(); // knots to m/s
+
+        if (window.gameController.game.option.get('simplifySpeeds') === 'no') {
+            this.updateGroundSpeedPhysics(scaleSpeed);
+        } else {
+            this.updateSimpleGroundSpeedPhysics(scaleSpeed);
+        }
+
+        this.distance = vlen(this.position);
+        this.radial = vradial(this.position);
+
+        if (this.radial < 0) {
+            this.radial += tau();
+        }
+
+        //TODO: I am not sure what this has to do with aircraft Physics
+        const isInsideAirspace = this.isInsideAirspace(window.airportController.airport_get());
```

Comment:
agreed.  these last few lines should live in a different method that gets called after this method.